algorithms/src/polycommit/kzg10/mod.rs

```diff
@@ -432,20 +443,22 @@ impl<E: PairingEngine> KZG10<E> {
         max_degree: usize,
         enforced_degree_bounds: Option<&[usize]>,
         p: impl Into<LabeledPolynomialWithBasis<'a, E::Fr>>,
-    ) -> Result<(), PCError> {
+    ) -> Result<()> {
         let p = p.into();
         if let Some(bound) = p.degree_bound() {
-            let enforced_degree_bounds = enforced_degree_bounds.ok_or(PCError::UnsupportedDegreeBound(bound))?;
+            let enforced_degree_bounds = enforced_degree_bounds
+                .ok_or_else(|| anyhow::anyhow!("The degree bound ({}) is not supported by the parameters", bound))?;
 
             if enforced_degree_bounds.binary_search(&bound).is_err() {
-                Err(PCError::UnsupportedDegreeBound(bound))
+                anyhow::bail!("The degree bound ({}) is not supported by the parameters", bound)
             } else if bound < p.degree() || bound > max_degree {
-                return Err(PCError::IncorrectDegreeBound {
-                    poly_degree: p.degree(),
-                    degree_bound: p.degree_bound().unwrap(),
-                    max_degree,
-                    label: p.label().to_string(),
-                });
+                anyhow::bail!(
+                    "The degree bound ({}) for the polynomial {} (having degree {}) is greater than the maximum degree ({})",
```

Comment:
This message doesn't reflect the condition "bound < p.degree". Perhaps I'm missing something? If not, maybe: (or even better, splitting this `else` into two separate cases)
```suggestion
                    "The degree bound ({}) for the polynomial {} is less than its degree ({}) or greater than the maximum degree ({})",
```